algorithms.tex

```diff
@@ -1412,6 +1412,15 @@
 behave as if they are implemented by calling \tcode{begin} and \tcode{end} on
 the \tcode{Range} and dispatching to the overload that takes separate
 iterator and sentinel arguments.
+
+\pnum
+\newtxt{Some algorithms declare both an overload that takes a \tcode{Range}
+and an \tcode{Iterator}, and an overload that takes two \tcode{Range}
+arguments. Since an array type~(\cxxref{basic.compound}) both satisifies
+\tcode{Range} and decays to a pointer~(\cxxref{conv.array}) which satisfies
+\tcode{Iterator}, such overloads are ambiguous when an array is passed as
+the second argument. Implementations shall provide a mechanism to resolve
+this ambiguity in favor of the overload that takes two ranges.}
```

Comment:
Yes, I've been pacing about the house considering pros and cons and have come to the same conclusion. Six months ago this would have been a great idea, but it's too much of a design change this late in the process. The change to the customization points was absolutely necessary to give them well-defined semantics; that is not the case for the algorithms.